src/ng/q.js

```diff
@@ -381,7 +381,11 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
       if (!toCheck.pur) {
         toCheck.pur = true;
         var errorMessage = 'Possibly unhandled rejection: ' + toDebugString(toCheck.value);
```

Comment:
Here is another idea (either for `toDebugString()` or just here):
If `toCheck.value` has `toString()` method that is not the default `Object.prototype.toString()`, then use that for serializing it. Something similar to [stringify][1].

[1]: https://github.com/angular/angular.js/blob/465d1734559ca4a7f4aa24387060f88fcc53ecb1/src/Angular.js#L1995